document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/document/description/impl/RetrieveApiEnumerationDescriptionImpl.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.companieshouse.document.generator.api.document.description.impl;
+
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.Yaml;
+import uk.gov.companieshouse.document.generator.api.document.RetrieveApiEnumerationDescription;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.document.generator.api.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
+
+@Service
+public class RetrieveApiEnumerationDescriptionImpl implements RetrieveApiEnumerationDescription {
+
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @Override
+    public String getApiEnumerationDescription(String fileName, String identifier, String accountType,
+                                               Map<String, String> parameters) throws IOException {
+
+        Yaml yaml = new Yaml();
+        File descriptionsFile = new File(fileName);
+
+        String description = "";
+
+        InputStream inputStream;
+        try {
+            inputStream = new FileInputStream(descriptionsFile);
```

Comment:
I would change this to a Try with resource so that you don't have to explicitly handle closing the Stream if an error occurs. So something like `try (InputStream inputStream = new FileInputStream(descriptionsFile)) {`